### PR TITLE
Add slurm 16.05

### PR DESCRIPTION
**Pull request by alarcher**
Slurm is a utility for scheduling jobs and allowing me to run nice simulations on my workstation.
I patched the latest stable but need help from SMF wizards for packaging it properly.

### CODE REVIEW

File: components/sysutils/slurm/files/getgrouplist.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*-
```
Comment (pyhalov) [inline components/sysutils/slurm/files/getgrouplist.h]: Why do we need it? Just to generate patches/11-getgrouplist.patch ? Then there's issue... getgrouplist.c is BSD-licensed, slurm - GPLv2-licensed. We can't mix such code in one file and I'm not sure if we can link such files in one object file.

Comment (alarcher) [inline components/sysutils/slurm/files/getgrouplist.h]: Right, I always forget that the relationship is unidirectional...
The reason is that illumos does not have getgrouplist() maybe I should file a RFE.

Comment (alarcher) [inline components/sysutils/slurm/files/getgrouplist.h]: BTW I took this patch from the Solaris port of KDE 4.

File: components/sysutils/slurm/patches/11-getgrouplist.patch [hunk 2]
```diff
@@ -0,0 +1,214 @@
+--- ./src/slurmd/slurmd/req.c.orig	2016-05-05 22:21:22.550661771 +0200
```
Comment (pyhalov) [inline components/sysutils/slurm/patches/11-getgrouplist.patch]: We can't do this... Licensing

Comment (alarcher) [inline components/sysutils/slurm/patches/11-getgrouplist.patch]: My understanding was that adding BSD-licensed code patches (or any permissive copyleft) to a GPL licensed code is not an issue while the converse is not possible: GPL code requires that the overall license is GPL.

Comment (pyhalov) [inline components/sysutils/slurm/patches/11-getgrouplist.patch]: http://www.gnu.org/licenses/gpl-faq.html#OrigBSD

Comment (alarcher) [inline components/sysutils/slurm/patches/11-getgrouplist.patch]: Oh sorry I did not see that it is the 4-Clause version.

Comment (alarcher) [inline components/sysutils/slurm/patches/11-getgrouplist.patch]: RFE "Add getgrouplist() to illumos": https://www.illumos.org/issues/7024

**Comment by alarcher** [comment]
Need to deliver example configuration files.

**Comment by xen0l** [comment]
Any news on this?

**Comment by alarcher** [comment]
Yes everything is fine, I fixed all the requests, I bumped to 16.05 while the original was 15.08 and also I fixed the getgrouplist patch issue using Alexander's pointer.
 I need to figure out how which generic default configuration files to deliver and I am not sure how they should look.

**Comment by pyhalov** [comment]
Any news on it?

**Comment by alarcher** [comment]
Some permissions to fix and the PR will be ready to be merged.

**Comment by alarcher** [comment]
Segmentation fault in slurmctld needs love...

**Comment by pyhalov** [comment]
Any news on it ? :) BTW, there's 17.02 version already...

**Comment by alarcher** [comment]
@pyhalov did they mention if there are more segfaults in the new version? ;) Could not figure out a working configuration yet, some weird segfaults.

**Pull request closed without merging**